### PR TITLE
CP-30433: add uefi_certificates field

### PR DESCRIPTION
**Pull request by rwdrich**

### CODE REVIEW

File: ocaml/idl/datamodel_common.ml [hunk 1]
```diff
@@ -8,7 +8,7 @@ open Datamodel_roles
               When introducing a new release, bump the schema minor version to the next hundred
               to leave a gap for potential hotfixes needing to increment the schema version.*)
 let schema_major_vsn = 5
-let schema_minor_vsn = 300
+let schema_minor_vsn = 301
```
Comment (edwintorok) [inline ocaml/idl/datamodel_common.ml]: Please make this 400 as the comment above indicates (this won't be part of Naples, therefore bump version by 100 to leave gap for fixes).

File: ocaml/idl/datamodel_host.ml [hunk 2]
```diff
@@ -1439,5 +1451,6 @@ let host_query_ha = call ~flags:[`Session]
            field ~qualifier:DynamicRO ~lifecycle:[Published, rel_falcon, ""] ~ty:(Set (Ref _feature)) "features" "List of features available on this host";
            field ~qualifier:StaticRO ~lifecycle:[Published, rel_kolkata, ""] ~default_value:(Some (VString "")) ~ty:String "iscsi_iqn" "The initiator IQN for the host";
            field ~qualifier:StaticRO ~lifecycle:[Published, rel_kolkata, ""] ~default_value:(Some (VBool false)) ~ty:Bool "multipathing" "Specifies whether multipathing is enabled";
+           field ~qualifier:StaticRO ~lifecycle:[Published, rel_naples, ""] ~default_value:(Some (VString "")) ~ty:String "uefi_certificates" "The UEFI certificates allowing Secure Boot"
```
Comment (edwintorok) [inline ocaml/idl/datamodel_host.ml]: This should be `rel_oslo`, which doesn't exist yet, you'll have to add one (and say unreleased, and just increment the version_minor).

File: ocaml/idl/datamodel_host.ml [hunk 3]
```diff
@@ -1268,6 +1268,17 @@ let host_query_ha = call ~flags:[`Session]
     ~allowed_roles:_R_POOL_OP
     ()
 
+  let set_uefi_certificates = call
+    ~name:"set_uefi_certificates"
+    ~lifecycle:[Published, rel_naples, ""]
```
Comment (edwintorok) [inline ocaml/idl/datamodel_host.ml]: oslo

File: ocaml/idl/datamodel_pool.ml [hunk 4]
```diff
@@ -745,5 +745,6 @@ open Datamodel_types
          ; field ~qualifier:RW ~in_product_since:rel_dundee ~default_value:(Some (VBool false)) ~ty:Bool "policy_no_vendor_device" "The pool-wide policy for clients on whether to use the vendor device or not on newly created VMs. This field will also be consulted if the 'has_vendor_device' field is not specified in the VM.create call."
          ; field ~qualifier:RW ~in_product_since:rel_ely ~default_value:(Some (VBool false)) ~ty:Bool "live_patching_disabled" "The pool-wide flag to show if the live patching feauture is disabled or not."
          ; field ~in_product_since:rel_inverness ~qualifier:DynamicRO ~ty:Bool ~default_value:(Some (VBool false)) "igmp_snooping_enabled" "true if IGMP snooping is enabled in the pool, false otherwise."
+; field ~in_product_since:rel_naples ~qualifier:RW ~ty:String ~default_value:(Some (VString "")) "uefi_certificates" "The UEFI certificates allowing Secure Boot"
```
Comment (edwintorok) [inline ocaml/idl/datamodel_pool.ml]: indentation? also oslo

File: ocaml/xapi/records.ml [hunk 5]
```diff
@@ -554,6 +554,7 @@ let pool_record rpc session_id pool =
       make_field ~name:"cpu_info" ~get:(fun () -> Record_util.s2sm_to_string "; " (x ()).API.pool_cpu_info) ~get_map:(fun () -> (x ()).API.pool_cpu_info) ();
       make_field ~name:"policy-no-vendor-device" ~get:(fun () -> string_of_bool (x ()).API.pool_policy_no_vendor_device) ~set:(fun s -> Client.Pool.set_policy_no_vendor_device rpc session_id pool (safe_bool_of_string "policy-no-vendor-device" s)) ();
       make_field ~name:"live-patching-disabled"  ~get:(fun () -> string_of_bool (x ()).API.pool_live_patching_disabled) ~set:(fun s -> Client.Pool.set_live_patching_disabled rpc session_id pool (safe_bool_of_string "live-patching-disabled" s)) ();
+      make_field ~name:"uefi-certificates" ~get:(fun () -> (x ()).API.pool_uefi_certificates) ()
```
Comment (edwintorok) [inline ocaml/xapi/records.ml]: Might want to set this one to `hidden:true`, it'll be giant blob of base64 encoded data. We wouldn't want to show it by default (see how the nvram field is marked)

File: ocaml/xapi/xapi_host.ml [hunk 6]
```diff
@@ -1825,6 +1826,16 @@ let allocate_resources_for_vm ~__context ~self ~vm ~live =
...
 
+(*let set_uefi_certificates ~rpc ~session_id ~host ~pool =
+  if Client.Client.Pool.get_uefi_certificates ~rpc ~session_id ~self:pool= "" then
+    Client.Client.Pool.set_uefi_certificates ~rpc ~session_id ~self:pool ~value:(Client.Client.Host.get_uefi_certificates ~rpc ~session_id ~self:host)*)
+
+let set_uefi_certificates ~__context ~host ~value =
+  Db.Host.set_uefi_certificates ~__context ~self:host ~value;
+  let pool = Helpers.get_pool ~__context in
+  if Db.Pool.get_uefi_certificates ~__context ~self:pool  = "" then
+    Db.Pool.set_uefi_certificates ~__context ~self:pool ~value:(Db.Host.get_uefi_certificates ~__context ~self:host)
```
Comment (edwintorok) [inline ocaml/xapi/xapi_host.ml]: You already have the `value` from above, no need to read it again.

File: ocaml/xapi/xapi_host.ml [hunk 7]
```diff
@@ -1825,6 +1826,16 @@ let allocate_resources_for_vm ~__context ~self ~vm ~live =
   (* Implemented entirely in Message_forwarding *)
   ()
 
+(*let set_uefi_certificates ~rpc ~session_id ~host ~pool =
+  if Client.Client.Pool.get_uefi_certificates ~rpc ~session_id ~self:pool= "" then
+    Client.Client.Pool.set_uefi_certificates ~rpc ~session_id ~self:pool ~value:(Client.Client.Host.get_uefi_certificates ~rpc ~session_id ~self:host)*)
```
Comment (edwintorok) [inline ocaml/xapi/xapi_host.ml]: this can be dropped?

**Comment by lindig** [comment]
Can we have ticket numbers as part of the commit subject unless these are really isolated small changes.

**Comment by edwintorok** [comment]
The xenserver-buildenv failure is because uefi branch was out of date on jenkins, I did a merge from master and trigerred another build.

**Pull request merged**